src/components/schduler/monthlyView/monthlyView.tsx

```diff
@@ -0,0 +1,161 @@
+  }
+
+  return weeks;
+};
+
+export const MonthlyView: React.FC<SchedulerTableProps> = ({
+  events,
+  className,
+}) => {
+  const [currentDate, setCurrentDate] = useState(new Date());
+
+  const handlePreviousMonth = () => {
+    setCurrentDate(
+      new Date(currentDate.getFullYear(), currentDate.getMonth() - 1, 1)
+    );
+  };
+
+  const handleNextMonth = () => {
+    setCurrentDate(
+      new Date(currentDate.getFullYear(), currentDate.getMonth() + 1, 1)
+    );
+  };
+
+  const year = currentDate.getFullYear();
+  const month = currentDate.getMonth();
+  const weeks = generateCalendar(year, month);
+
+  const monthEvents = events.filter((event) => {
+    const eventStart = new Date(event.startTime);
+    const eventEnd = new Date(event.endTime);
+    return (
+      (eventStart.getFullYear() === year && eventStart.getMonth() === month) ||
+      (eventEnd.getFullYear() === year && eventEnd.getMonth() === month)
+    );
+  });
+
+  return (
+    <div className={`${className} p-4`}>
+      <div className="flex justify-between mb-4">
+        <button
+          onClick={handlePreviousMonth}
+          className="px-4 py-2 bg-primaryVariant rounded "
+        >
+          <span className="cursor-pointer text-primary ">
+            <FontAwesomeIcon icon={faChevronLeft} />
+          </span>
+        </button>
+        <h2 className="text-xl font-semibold text-primary">
+          {currentDate.toLocaleString("default", {
+            month: "long",
+            year: "numeric",
+          })}
+        </h2>
+        <button
+          onClick={handleNextMonth}
+          className="px-4 py-2 bg-primaryVariant rounded "
+        >
+          <span className="cursor-pointer text-primary">
+            <FontAwesomeIcon icon={faChevronRight} />
+          </span>
+        </button>
+      </div>
+      <table className="table-auto w-full border-collapse">
+        <thead>
+          <tr>
+            {[
+              "Sunday",
+              "Monday",
+              "Tuesday",
+              "Wednesday",
+              "Thursday",
+              "Friday",
+              "Saturday",
+            ].map((day) => (
+              <th key={day} className="border border-gray-200 p-2 min-w-28">
+                {day}
+              </th>
+            ))}
+          </tr>
+        </thead>
+        <tbody>
+          {weeks.map((week, i) => (
+            <tr key={i}>
+              {week.map((day, j) => (
+                <td
+                  key={j}
+                  className="border border-gray-200 p-2 h-24 align-top"
+                >
+                  {day && (
+                    <>
+                      <div className="text-sm font-semibold">
+                        {day.getDate()}
+                      </div>
+                      {monthEvents
+                        .filter(
+                          (event) =>
+                            new Date(event.startTime).toDateString() ===
```

Comment:
It won`t work as it filters the events that exactly match the day string so that if an event happens on the same day at 13:00 the iso string will differ from the day iso string

consider matching the first part of the iso string or just match the month and day of each event